documentation/design-docs/diagnostics-client-library.md

```diff
@@ -0,0 +1,298 @@
+# Diagnostics Client Library API Design 
+
+## Intro
+The Diagnostics Client Library (currently named as "Runtime Client Library") is a managed library that can be used to interact with the .NET runtime via the diagnostics IPC protocol as documented in https://github.com/dotnet/diagnostics/blob/master/documentation/design-docs/ipc-protocol.md.
+
+The name "Diagnostics Client Library" comes from the fact that we call the runtime (CoreCLR) component responsible for accepting and handling the diagnostics IPC commands the "diagnostics server" - https://github.com/dotnet/coreclr/blob/master/src/vm/diagnosticserver.h. Since this library is a managed library on the other side of the IPC protocol responsible for communicating with the runtime's "diagnostics server", calling this the "Diagnostics Client Library" made sense.
```

Comment:
Microsoft.CoreCLR.DiagnosticsClient.dll sounds good to me as the assembly name. I’ll add that to the doc as well.

Reply:
It isn't consist with the rest of our assemblies. They all start with Microsoft.Diagnostics... And having "CoreCLR" in the name seems like an internal name for .NET Core.  What about  Microsoft.Diagnostics.NETCore.Client?

Reply:
How about Microsoft.NETCore.Diagnostics.Client?

Reply:
With the branding change of “.NET Core” -> “.NET” for 5.0, maybe we should avoid using “core” in the name.

Reply:
The brand name might be changing to .NET but the diagnostics client only talks to CoreCLR and not any other .NET runtimes. (i.e. Mono) - unless supporting that is also a part of our plan..? I'm not aware of such plan though.

Reply:
IMO It's a .NET Core library, I do not see a reason to add the `NETCore` to the namespace.

Reply:
I'm still inclined to put either NETCore or CoreCLR because just saying 'NET' seems too general and potentially misleading since this isn't compatible with other .NET runtimes.

Reply:
You should write the name of the assembly into the document : ) Its not always clear from the discussion what the plan of record is.